centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -156,64 +167,170 @@ function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
                 $statement->execute();
                 $statement = $pearDB->query("SELECT MAX(sc_id) as maxid FROM `service_categories`");
                 $maxId = $statement->fetch();
-                $scAcl[$maxId['maxid']] = $key;
-                $query = "INSERT INTO service_categories_relation (service_service_id, sc_id) " .
-                    "(SELECT service_service_id, :max_id" .
-                    " FROM service_categories_relation WHERE sc_id = :sc_id)";
-                $statement = $pearDB->prepare($query);
-                $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-                $statement->bindValue(':max_id', $maxId['maxid'], \PDO::PARAM_INT);
-                $statement->execute();
+
+                if (isset($maxId['maxid'])) {
+                    $scAcl[$maxId['maxid']] = $scId;
+                    try {
+                        $selectServiceIdsStatement = $pearDB->prepareQuery(
+                            <<<SQL
+                                SELECT service_service_id FROM service_categories_relation
+                                WHERE sc_id = :sc_id
+                            SQL
+                        );
+                        $pearDB->executePreparedQuery($selectServiceIdsStatement, ['sc_id' => $scId]);
+                        $insertNewRelationStatement = $pearDB->prepareQuery(
+                            <<<SQL
+                                INSERT INTO service_categories_relation (service_service_id, sc_id)
+                                VALUES (:serviceId, :maxId)
+                            SQL
+                        );
+                        $foundServiceIds = [];
+                        while ($serviceId = $pearDB->fetchColumn($selectServiceIdsStatement)) {
+                            $pearDB->executePreparedQuery($insertNewRelationStatement, [
+                                'serviceId' => $serviceId,
+                                'maxId' => $maxId['maxid']
+                            ]);
+                            $foundServiceIds[] = $serviceId;
+                        }
+                        if (! empty($foundServiceIds)) {
+                            $fields["sc_services"] = implode(", ", $foundServiceIds);
+                        }
+
+                        $centreon->CentreonLogAction->insertLog(
+                            object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+                            object_id: $maxId['maxid'],
+                            object_name: $sc_name,
+                            action_type: ActionLog::ACTION_TYPE_ADD,
+                            fields: $fields
+                        );
+                    } catch (CentreonDbException $ex) {
+                        CentreonLog::create()->error(
+                            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+                            message: $ex->getMessage(),
```

Comment:
Not only the message of the exception, it's not a good practice, you need context too.

Reply:
Can you provide an example of context

Reply:
good